### PR TITLE
Timeout for GATTServer's connect, disconnect methods.

### PR DESCRIPTION
**Pull request by zakorgy**
Added 30 seconds maximum timeout as specified in the bluetooth 4.2 https://www.bluetooth.org/DocMan/handlers/DownloadDoc.ashx?doc_id=286439 (Vol. 3, page 480). With this the existing `bluetooth_device_disconnect.html` test will work correctly. It will not show the `connected:true` message, before the connection established, and will not show the `connected: false` message, before not disconnected from the GATT server.

### CODE REVIEW

File: components/net/bluetooth_thread.rs [hunk 1]
```diff
@@ -33,6 +33,11 @@ const CHARACTERISTIC_ERROR: &'static str = "No characteristic found";
 const DESCRIPTOR_ERROR: &'static str = "No descriptor found";
 const VALUE_ERROR: &'static str = "No characteristic or descriptor found with that id";
 const SECURITY_ERROR: &'static str = "The operation is insecure";
+const NETWORK_ERROR: &'static str = "A network error occurred";
+// A transaction not completed within 30 seconds shall time out. Such a transaction shall be considered to have failed.
+// https://www.bluetooth.org/DocMan/handlers/DownloadDoc.ashx?doc_id=286439 (Vol. 3, page 480)
+const MAXIMUM_TARNSACTION_TIME: u32 = 30;
```
Comment (mmatyas) [inline components/net/bluetooth_thread.rs]: typo

File: components/net/bluetooth_thread.rs [hunk 2]
```diff
@@ -33,6 +33,11 @@ const CHARACTERISTIC_ERROR: &'static str = "No characteristic found";
 const DESCRIPTOR_ERROR: &'static str = "No descriptor found";
 const VALUE_ERROR: &'static str = "No characteristic or descriptor found with that id";
 const SECURITY_ERROR: &'static str = "The operation is insecure";
+const NETWORK_ERROR: &'static str = "A network error occurred";
+// A transaction not completed within 30 seconds shall time out. Such a transaction shall be considered to have failed.
```
Comment (mmatyas) [inline components/net/bluetooth_thread.rs:37]: Isn't this line too long? What does the style say?

Comment (zakorgy) [inline components/net/bluetooth_thread.rs:37]: No, it ends at 120 :)

**Comment by mmatyas** [comment]
looks good!

**Comment by dati91** [comment]
lgtm

**Pull request merged**